### ISSUE TITLE
Abema TV support

### ISSUE THREAD

**Issue by saligondal** [ytdl-org/youtube-dl#33002]
## Checklist

- [x ] I'm reporting a new site support request
- [x ] I've verified that I'm running youtube-dl version **2021.12.17**
- [ ] I've checked that all provided URLs are alive and playable in a browser
- [ ] I've checked that none of provided URLs violate any copyrights
- [ ] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- Single video: https://abema.tv/channels/fighting-sports2/slots/C3K1gmnm9FchM1?lang=en

## Description

https://abema.tv/channels/fighting-sports2/slots/C3K1gmnm9FchM1?lang=en
trying to download this video but i get this error:

ERROR: No video formats found; please report this issue on https://github.com/ytdl-org/youtube-dl/issues , using the appropriate issue template. Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose option and include the complete output.

**Comment by dirkf** [comment]
Feel free to complete the checklist in order to progress your request.  Also, perhaps first, check that you can play the suggested video in a browser with DRM disabled.

**Comment by saligondal** [comment]
disabled drm and video plays fine on chrome

**Comment by dirkf** [comment]
There is an extractor for this site that ought to handle the problem page in _yt-dlp_. The extractor uses exceptional techniques to handle the playback licence request, interfacing to parts of the core program that are now somewhat differently implemented.

My back-port of the _yt-dlp_ extractor is now behaving like the original, but presumably due to geo-restriction they both fail (403) on fetching the media manifest, and the playback stage is never reached, so I can't tell if my version of the abematv-license:// handler is actually working.

For this site to be supported here and not just in _yt-dlp_, someone in-region, ideally with access to an account, will have to step up to test and/or further develop a PR version that I could post.

**Comment by adityakrmishra** [comment]
Hey @dirkf  and @saligondal  I'd like to take a crack at this.

@dirkf, regarding the 403 geo-restriction block on the media manifest: I have a reliable Japanese VPN set up, so I should be able to bypass the regional block and properly test the playback license request from within the region.

My plan is to backport the abematv.py extractor from yt-dlp, adapting their abematv-license:// handler and specific API calls to fit youtube-dl's core architecture. I'll run some local tests via the VPN and see if I can successfully pull the manifest.

I'll report back here with my findings or open a PR if I get a working prototype going.

**Comment by adityakrmishra** [comment]
@dirkf  I've fully backported the abematv.py extractor, including the abematv-license:// handler, custom headers, and AES crypto primitives. However, my VPN IP is being hard-blocked (403) by Akamai at the m3u8 CDN level, so I can't test the final fragment download. I'm opening a Draft PR with the backported code—could someone with a clean Japanese residential IP give it a test run?

**Comment by dirkf** [comment]
Is it your VPN, or is it another case where _yt-dlp_ wins using impersonation?

This is what my WIP (from 2025-03) gets with a free and so probably ineffective JP VPN:
```console
$ python -m youtube_dl -vF 'https://abema.tv/channels/fighting-sports2/slots/C3K1gmnm9FchM1?lang=en'
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-vF', u'https://abema.tv/channels/fighting-sports2/slots/C3K1gmnm9FchM1?lang=en']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Git HEAD: 71ac16cbc
[debug] Python 2.7.15 (CPython i686 32bit) - Linux-6.1.0-40-686-pae-i686-with-debian-12.13 - OpenSSL 1.1.1a  20 Nov 2018 - glibc 2.1.3
[debug] exe versions: ffmpeg 5.1.8-0, ffprobe 5.1.8-0
[debug] Proxy map: {}
[AbemaTV] Extracting URL: https://abema.tv/channels/fighting-sports2/slots/C3K1gmnm9FchM1?lang=en
[AbemaTV] Authorizing
[AbemaTV] C3K1gmnm9FchM1: Downloading webpage
[AbemaTV] C3K1gmnm9FchM1: Checking playability
[AbemaTV] C3K1gmnm9FchM1: Downloading m3u8 information
ERROR: [AbemaTV] **???**: Failed to download m3u8 information: HTTP Error 403: Forbidden (caused by HTTPError()); please report this issue on https://github.com/ytdl-org/youtube-dl/issues , using the appropriate issue template. Make sure you are using the latest version; see  https://github.com/ytdl-org/youtube-dl/#user-content-installation  on how to update. Be sure to call youtube-dl with the --verbose option and include the complete output.
  File "youtube_dl/extractor/common.py", line 743, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "youtube_dl/YoutubeDL.py", line 2598, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/home/df/miniconda3/lib/python2.7/urllib2.py", line 435, in open
    response = meth(req, response)
  File "/home/df/miniconda3/lib/python2.7/urllib2.py", line 548, in http_response
    'http', request, response, code, msg, hdrs)
  File "/home/df/miniconda3/lib/python2.7/urllib2.py", line 473, in error
    return self._call_chain(*args)
  File "/home/df/miniconda3/lib/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "/home/df/miniconda3/lib/python2.7/urllib2.py", line 556, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)

```

**Comment by adityakrmishra** [comment]
@dirkf  I suspect it is absolutely the lack of impersonation.

During my testing, I explicitly injected the exact headers that yt-dlp uses for the manifest request (User-Agent matching the Android app, Origin, Referer, and Authorization: Bearer <token>). Even with a premium Japanese VPN and perfect headers, the Akamai CDN still threw a 403.

Since yt-dlp routes its requests through an impersonation layer (mimicking modern browser TLS fingerprints), it slips past Akamai's WAF. With youtube-dl relying on standard urllib, our TLS Client Hello is highly recognizable. Without a native impersonation layer in youtube-dl, this CDN block might be a hard wall for the Abema manifest fetch.

(Also, I noticed your terminal output has the [AbemaTV] **???**: error prefix—awesome to see the core error formatting patch in action locally!)

**Comment by dirkf** [comment]
>(Also, I noticed your terminal output has the [AbemaTV] ???: error prefix—awesome to see the core error formatting patch in action locally!)

And with the `get_temp_id()` version:
```console
...
$ python -m youtube_dl -vF 'https://abema.tv/channels/fighting-sports2/slots/C3K1gmnm9FchM1?lang=en'
[debug] System config: []
...
[AbemaTV] Extracting URL: https://abema.tv/channels/fighting-sports2/slots/C3K1gmnm9FchM1?lang=en
[AbemaTV] Authorizing
[AbemaTV] C3K1gmnm9FchM1: Downloading webpage
[AbemaTV] C3K1gmnm9FchM1: Checking playability
[AbemaTV] C3K1gmnm9FchM1: Downloading m3u8 information
ERROR: [AbemaTV] C3K1gmnm9FchM1: Failed to download m3u8 information: HTTP Error 403: Forbidden (caused by HTTPError()); please report this issue on https://github.com/ytdl-org/youtube-dl/issues , using the appropriate issue template. Make sure you are using the latest version; see  https://github.com/ytdl-org/youtube-dl/#user-content-installation  on how to update. Be sure to call youtube-dl with the --verbose option and include the complete output.
  File "youtube_dl/extractor/common.py", line 743, in _request_webpage
...
```

Anyhow, as I suspected, this is probably blocked by #33050.

**Comment by adityakrmishra** [comment]
@dirkf  Ah, that makes perfect sense. If it's blocked by the lack of TLS impersonation in #33050, then there's no way around Akamai's WAF with standard urllib.

I'll leave the Draft PR (#33270) open and parked. Once #33050 gets resolved and an impersonation layer lands in core, we can easily hook this extractor up to it and finally pull those manifests.

(And seeing that ```C3K1gmnm9FchM1``` ID successfully parsed in the error output is awesome!)
